src/webserver/DOSGuard.h

```diff
@@ -9,11 +9,47 @@
 class DOSGuard
 {
     boost::asio::io_context& ctx_;
-    std::mutex mtx_;  // protects ipFetchCount_
+    std::recursive_mutex mtx_;  // protects ipFetchCount_, ipRequestCount_
```

Comment:
Would you make a note about why you chose `recursive_mutex`? I know we discussed that it allows people to directly call `getRequest` or `getFetches` outside of `checkout`, but just make a comment.